src/libopensc/card-openpgp.c

```diff
@@ -664,13 +697,35 @@ pgp_parse_algo_attr_blob(const pgp_blob_t *blob, sc_cardctl_openpgp_keygen_info_
 				return SC_ERROR_INCORRECT_PARAMETERS;
 
 			key_info->algorithm = SC_OPENPGP_KEYALGO_RSA;
-			key_info->rsa.modulus_len = bebytes2ushort(blob->data + 1);
-			key_info->rsa.exponent_len = bebytes2ushort(blob->data + 3);
+			key_info->u.rsa.modulus_len = bebytes2ushort(blob->data + 1);
+			key_info->u.rsa.exponent_len = bebytes2ushort(blob->data + 3);
 
-			key_info->rsa.keyformat = (blob->len > 5)
+			key_info->u.rsa.keyformat = (blob->len > 5)
 						  ? blob->data[5]
 						  : SC_OPENPGP_KEYFORMAT_RSA_STD;
 			break;
+		case SC_OPENPGP_KEYALGO_ECDH:
+		case SC_OPENPGP_KEYALGO_ECDSA:
+
+			/* SC_OPENPGP_KEYALGO_ECDH || SC_OPENPGP_KEYALGO_ECDSA */
+			key_info->algorithm = blob->data[0];
+
+			/* Create copy of oid from blob */
+			for (j=0; j < (blob->len-1); j++) {
+				oid.value[j] = blob->data[j+1]; /* ignore first byte of blob (algo ID) */
+				if (j == blob->len-2)
+					oid.value[j+1] = -1;
```

Comment:
I'm just curious: Why are you setting the last byte to 0xFF, why aren't you intializing the rest of `oid.value` to 0xFF?

Reply:
Just because I didn't think about this solution. I am using sc_init_oid instead now.